src/client/src/components/Quiz/Quiz.jsx

```diff
@@ -1,12 +1,28 @@
 import "./quiz.css";
 import BasicQuestion from "./BasicQuestion/BasicQuestion";
 import ProgressBar from "./ProgressBar/ProgressBar";
+import { useEffect, useState } from "react";
+
+export default function Quiz({ questions, removeQuestion }) {
+  const [question, setQuestion] = useState(questions[0]);
+
+  useEffect(() => {
+    if (questions.length != 1) {
+      setQuestion(questions[0]);
+    }
+  }, [questions]);
 
-export default function Quiz() {
   return (
     <div className="quiz-container">
-      <ProgressBar />
-      <BasicQuestion />
+      <ProgressBar progressPercentage={100 - questions.length} />
```

Comment:
questions.length is calculated not in %, right? 

and the answered question id deleted from the questions array?

so, in this case if you answered 6 questions out of 10, you have 4 questions left, it is questions.length. And the formula gives you 100 - 4 = 96?

Maybe we need to calculate`amountOfAnsweredQuestions / amountOfAllQuestions* 100 `? If you answered 6 questions out of 10 then  6 / 10 * 100 = 60%.